jdaviz/configs/mosviz/plugins/parsers.py

```diff
@@ -463,14 +463,28 @@ def mos_meta_parser(app, data_obj, ids=None, spectra=False, sp1d=False):
             _add_to_table(app, dec, "Dec.")
 
 
-def _get_object_keyword(data_obj, ids):
-    if ids is not None:
-        # remove leading path to file name
-        ids_short = [os.path.basename(d) for d in ids]
-        names = [x[0].header.get("OBJECT", d) for x, d in zip(data_obj, ids_short)]
-    else:
-        names = [x[0].header.get("OBJECT", FALLBACK_NAME) for x in data_obj]
-    return names
+def _get_source_names(hdus, filepaths=None):
+    """
+    Attempts to extract a list of source identifiers via different header values.
+
+    Parameters
+    ----------
+    hdus : a list of HDUs (NOT an HDUList!) to check the header of. Filtering should
+        be done in advance to decide which HDUs to check against (e.g. Sci only)
+    filepaths : Optional. A list of filepaths to fallback on if no header values are
+        identified
+    """
+    # Try to extract the Source ID:
+    try:
+        src_names = ["Source ID: " + str(x.header['SOURCEID']) for x in hdus]
+    except KeyError:
+        if filepaths:
+            # remove leading path to file name
+            filenames = [os.path.basename(d) for d in filepaths]
```

Comment:
This might also work (untested).
```suggestion
            filenames = list(map(os.path.basename, filepaths))
```

Reply:
Stale